crates/dojo-physics/src/bool/bvec2.cairo

```diff
@@ -0,0 +1,31 @@
+struct BVec2Type {
+    x: bool,
+    y: bool,
+}
+
+// Traits for BVec2Type
+trait BVec2 {
+    // Constructors
+    fn new(x: bool, y: bool) -> BVec2Type;
+    fn splat(v: bool) -> BVec2Type;
```

Comment:
We should use generics rather than defining a type for each

```suggestion
struct Vec2<T> {
    x: T,
    y: T,
}

// Traits for Vec2
trait Vec2<T> {
    // Constructors
    fn new(x: T, y: T) -> Vec2;
    fn splat(v: T) -> Vec2;
```

Reply:
you can find an example here: https://github.com/keep-starknet-strange/quaireaux/blob/main/quaireaux/data_structures/src/queue.cairo

Reply:
is it ok to use `struct Vec2Type<T>` instead, to use same pattern as in cubit?